internal/seed/seed.go

```diff
@@ -173,5 +187,5 @@ func (p Node) WithStatus(status Status) Node {
 }
 
 func (p Node) Healthy() bool {
-	return p.Status.CatchingUp || !p.Status.Reachable || !p.Status.IsLatestBlock
+	return !p.Status.CatchingUp && p.Status.Reachable && p.Status.IsLatestBlock
 }
```

Comment:
This was implemented as an "unhealthy" check, made the correction on the implementation and on the usage.